core/keyboard_nav/key_action.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2019 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview The class representing a keyboard action.
+ * Used primarily for keyboard navigation.
+ */
+'use strict';
+
+goog.provide('Blockly.KeyAction');
+
+/**
+ * Class for a single action.
+ * There can be one action for each key. If the action only applies to a
+ * single state (toolbox, flyout, workspace) then the function should handle this.
+ * @param {string} name The name of the action.
+ * @param {string} desc The description of the action.
+ * @param {string} func The function to be called when the key is pressed.
```

Comment:
This is a {function} not a {string}.

Reply:
{Function}*

Reply:
It seems to appear both ways in code, is there a preference?

Reply:
https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System

> Also note the difference between `function()` and `Function`. The latter is an instance type and is nullable by default. `function(...)` should be used instead of `Function` whenever possible because it provides more type information about its parameters and return value.